discord/ext/commands/cooldowns.py

```diff
@@ -48,21 +48,26 @@ def __init__(self, rate, per, type):
         if not isinstance(self.type, BucketType):
             raise TypeError('Cooldown type must be a BucketType')
 
-    def is_rate_limited(self):
+    def is_rate_limited(self, current=None):
+        if not current:
+            current = time.time()
+
+        # It's fine to update here, as long as we never create a new window
+        # so check if our window has passed and we can refresh our tokens
+        if current > self._window + self.per:
```

Comment:
Since this a public API that apparently takes a `current` parameter, updating is actually not a good idea.

Consider: Someone passing a time.time() 3 minutes in the future.

It would probably be best to just move this updating in the `update_rate_limit` call and use a local `tokens` variable here that defaults to `self._token`.